postprocess.py

```diff
@@ -92,51 +94,61 @@ def parse_args():
 # Load intervals
 logger.info('Loading intervals')
 intervals = pd.read_csv(args.intervals_file, sep='\t', header=None)
-
 # Get batch parameters
 with h5py.File(args.predictions_file, 'r') as infile:
     num_batches = infile['data'].shape[0]
     interval_size = infile['data'].shape[1]
 
-# TODO: assert that len(intervals) equals number of intervals in predictions file
-# TODO: assert that size of intervals in intervals file equals size of intervals in predictions file
+if ((int(intervals[2][0]) - int(intervals[1][0])) != interval_size):
+    logger.info("Interval sizes do not match between inference file and the intervals file")
+    sys.exit(0)
 
 # Load predictions, convert to bedGraph and append for each batch
 logger.info(
-    'Writing scored intervals to bedGraph file {}.bedGraph'.format(args.prefix))
+    'Writing scored intervals to bigwig file {}.bw'.format(args.prefix))
+
+def df_to_bigwig(batch_data, bwoutputfilename):
+    sizetuple = []
+    uniq_chroms = batch_data.chrom.unique()
+    with open(args.sizes_file, "r") as sizefile:
+        sizes  = sizefile.readlines()
+        for size in sizes:
+            size = size.strip().split("\t")
+            if size[0] in uniq_chroms:
+                sizetuple.append((size[0], int(size[1])))
+    bw = pyBigWig.open(bwoutputfilename, "w")
```

Comment:
this generate a .bw file per batch. there's also a `combiner` function that then combines all the intermediate files per batch into a single `.bw`. Does that work okay for multiple .bw files? Since each file has it's own header I'm wondering how well that's being handled in the combiner

Reply:
Good catch. It's doing a fine job as far as I can tell. I can open the file once and send in the object if that's a better coding practice.